File: test_data/code/java/1869-longer-contiguous-segments-of-ones-than-zeros.java
class Solution {
    public boolean checkZeroOnes(String s) {
        int maxOnes = 0, maxZeros = 0, currentOnes = 0, currentZeros = 0;
        
        for (char c : s.toCharArray()) {
            if (c == '1') {
                currentOnes++;
                maxOnes = Math.max(maxOnes, currentOnes);
                currentZeros = 0;
            } else {
                currentZeros++;
                maxZeros = Math.max(maxZeros, currentZeros);
                currentOnes = 0;
            }
        }
        
        return maxOnes > maxZeros;
    }
}

File: test_data/code/python/1869-longer-contiguous-segments-of-ones-than-zeros.py
Python
class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        maxOnes = 0
        maxZeros = 0
        currentOnes = 0
        currentZeros = 0

        for char in s:
            if char == '1':
                currentOnes += 1
                maxOnes = max(maxOnes, currentOnes)
                currentZeros = 0
            else:
                currentZeros += 1
                maxZeros = max(maxZeros, currentZeros)
                currentOnes = 0

        return maxOnes > maxZeros